### PR TITLE
Fix server side errors

### PR DESCRIPTION
**Pull request by menghanl**
#873
 * Service handler should get RPC errors rather than StreamError or ConnectionError
 * When context timeout happens, service handler should get deadline exceeded error (not true if connection is disconnected before context timeout)

### CODE REVIEW

**Comment by tamird** [comment]
does `transport.StreamErrorf` want to be unexported?

**Comment by menghanl** [comment]
Yes, we should! Done.

File: rpc_util_test.go [hunk 1]
```diff
@@ -173,8 +174,8 @@ func TestContextErr(t *testing.T) {
 		// outputs
 		errOut transport.StreamError
 	}{
-		{context.DeadlineExceeded, transport.StreamErrorf(codes.DeadlineExceeded, "%v", context.DeadlineExceeded)},
-		{context.Canceled, transport.StreamErrorf(codes.Canceled, "%v", context.Canceled)},
+		{context.DeadlineExceeded, transport.StreamError{codes.DeadlineExceeded, fmt.Sprintf("%v", context.DeadlineExceeded)}},
```
Comment (tamird) [inline rpc_util_test.go]: not too important, but you can use `.Error()` instead of `fmt.Sprintf("%v", ...)` here and on the next line

Comment (menghanl) [inline rpc_util_test.go]: Done.

**Pull request merged**